projecttacoma/deqm-test-server-frontend
# Care gaps components

## Summary
Implementation of Care Gaps button on Measure home page and Care Gaps page render with input components. 

**NOTE:** There are placeholders (anything under the Program text input component) rendered for formatting purposes that are not part of the task and need not be evaluated at this time. Some state variables are also used but not set as a result.

## New behavior
- A "Calculate Care Gaps" button is rendered on any Measure's home page (in between the "Evaluate Measure" and "Delete" buttons).
- If the "Calculate Care Gaps" button is pressed, user is redirected to the /Measure/[id]/careGaps page. 
- Where the following are rendered: 2 pre-filled DatePickers, 2 radio buttons (Select, Organization), AutoComplete components for Select Patient, Organization, and Practitioner, TextInput component for Program, and Placeholders for the request preview and calculate button

## Code changes
- **pages/[resource]/[id]/careGaps.tsx** added, renders the new page and components
- **__tests__/pages/resource/id/careGaps.tsx** added with unit tests

## Testing Guidance
- See the README.md for first time setup instructions.
#### Test in a browser
- To start the frontend: `npm run dev` then navigate to http://localhost:3001 in a browser
1. Valid navigation:
    - Click on “Measure” from the left-hand-side navigation menu. This will load the measure resource ID page. Click on any of the IDs. 
    - Click the “Calculate Care Gaps” button that is at the top of the main page body and to the left of the “Delete” button. User should be redirected to /Measure/[id]/careGaps page, where the following is rendered: 2 pre-filled DatePickers, 2 radio buttons (Select, Organization), AutoComplete components for Select Patient, Organization, and Practitioner, TextInput component for Program, and Placeholders for the request preview and calculate button.
      - The Subject radio button should be selected by default, and the Select Patient AutoComplete component will be required. The Select Organization and Select Practitioner components will be disabled.
      - Select the Organization radio button. Then the Select Organization and Select Practitioner components will be enabled (Select Organization will be required). The Select Patient component will be disabled.
    - User will be able to type into any enabled text/DatePicker components
2. Invalid navigation (non-Measure resource): 
    - Navigate to any [resourceType]/[id] page for a non-Measure resource. Then add "careGaps" to the end of the url and hit enter. An error message should be displayed on the main page body:  "Cannot calculate care gaps on resourceType: [resourceType], only on resourceType: Measure"

#### Unit tests
- Run the unit tests: `npm run test`
